File: Data Structures & Algorithms/two-integer-sum/submission-7.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> index;

        for(int i = 0; i < nums.size(); i++){
            index[nums[i]] = i;
        }

        for(int i = 0; i < nums.size(); i++){
            int ans = target - nums[i];
            if(index.count(ans) && index[ans] != i){
                return {i, index[ans]};
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = set(nums)
        length = 0
        for num in ans:
            if (num - 1) not in ans:
                longest = 1
                while num + longest in ans:
                    longest += 1
                length = max(length, longest)
        return length